### ISSUE TITLE
the top-5acc of validation set is greater than 1

### ISSUE THREAD

**Issue by xiaozhucj** [open-mmlab/mmaction2#2136]
Before raising a question, you may need to check the following listed items.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmaction2.readthedocs.io/en/latest/faq.html) but cannot get the expected help.

**Comment by xiaozhucj** [comment]
**Describe the issue**
When I used i3d for training, the model did not evaluate, but only output the loss value. When performing verification, the top-k accuracy of the verification set is abnormal, and the top-5acc is greater than 1
![image](https://user-images.githubusercontent.com/90119603/207309691-2c37392d-1376-4523-be78-7e6ec23a4e1f.png)
![image](https://user-images.githubusercontent.com/90119603/207309723-d2cf0ad6-671a-4a9f-94e8-9852a842526b.png)


**Reproduction**
1、What config dir you run?
	configs/recognition/i3d/i3d_r50_video_imgaug_32x2x1_100e_kinetics400_rgb.py
1.	What dataset do you use?
I made a action recognition dataset. The annotation format refers to the videoDataset in the framework
Also, modified the config file to use this video format.

`_base_ = ['../../_base_/models/i3d_r50.py']
# dataset settings
dataset_type = 'VideoDataset'
data_root = '/data/datasets/data/train/'
data_root_val = '/data/datasets/data/val/'
ann_file_train = '/data/datasets/data/train.txt'
ann_file_val = '/data/datasets/data/val.txt'
ann_file_test = '/data/datasets/data/test.txt'

img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_bgr=False)
train_pipeline = [
    dict(type='OpenCVInit'),
    dict(type='SampleFrames', clip_len=32, frame_interval=2, num_clips=1),
    dict(type='OpenCVDecode'),
    dict(type='Resize', scale=(-1, 256)),
    dict(
        type='MultiScaleCrop',
        input_size=224,
        scales=(1, 0.8),
        random_crop=False,
        max_wh_scale_gap=0),
    dict(type='Resize', scale=(224, 224), keep_ratio=False),
    dict(
```
        type='Imgaug',
        transforms=[
            dict(type='Fliplr', p=0.5),
            dict(type='Rotate', rotate=(-20, 20)),
            dict(type='Dropout', p=(0, 0.05))
        ]),
    # dict(type='Imgaug', transforms='default'),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='FormatShape', input_format='NCTHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs', 'label'])
]
val_pipeline = [
    dict(type='OpenCVInit'),
    dict(
        type='SampleFrames',
        clip_len=32,
        frame_interval=2,
        num_clips=1,
        test_mode=True),
    dict(type='OpenCVDecode'),
    dict(type='Resize', scale=(-1, 256)),
    dict(type='CenterCrop', crop_size=224),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='FormatShape', input_format='NCTHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs'])
]
test_pipeline = [
    dict(type='OpenCVInit'),
    dict(
        type='SampleFrames',
        clip_len=32,
        frame_interval=2,
        num_clips=1,
        test_mode=True),
    dict(type='OpenCVDecode'),
    dict(type='Resize', scale=(-1, 256)),
    dict(type='ThreeCrop', crop_size=256),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='FormatShape', input_format='NCTHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs'])
]
data = dict(
    videos_per_gpu=16,
    workers_per_gpu=4,
    val_dataloader=dict(videos_per_gpu=1),
    test_dataloader=dict(videos_per_gpu=1),
    train=dict(
        type=dataset_type,
        ann_file=ann_file_train,
        data_prefix=data_root,
        pipeline=train_pipeline),
    val=dict(
        type=dataset_type,
        ann_file=ann_file_val,
        data_prefix=data_root_val,
        pipeline=val_pipeline),
    test=dict(
        type=dataset_type,
        ann_file=ann_file_val,
        data_prefix=data_root_val,
        pipeline=test_pipeline))
# optimizer
evaluation = dict(
    interval=1, metrics=['top_k_accuracy'])

work_dir = './work_dirs/i3d_r50_video_3d_32x2x1_100e_kinetics400_rgb/'
optimizer = dict(
    type='SGD', lr=0.000625, momentum=0.9,
    weight_decay=0.0001)  # this lr is used for 4 gpus
optimizer_config = dict(grad_clip=dict(max_norm=40, norm_type=2))
# learning policy
lr_config = dict(
    policy='step',
    step=[15, 30],
    warmup='linear',
    warmup_ratio=0.1,
    warmup_by_epoch=True,
    warmup_iters=16)

total_epochs = 58
checkpoint_config = dict(interval=4)
workflow = [('train', 1)]

log_config = dict(
    interval=20,
    hooks=[
        dict(type='TextLoggerHook'),
        #    dict(type='TensorboardLoggerHook'),
    ])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
find_unused_parameters = False`



**Environment**
sys.platform: linux
Python: 3.7.13 (default, Mar 29 2022, 02:18:16) [GCC 7.5.0]
CUDA available: True
GPU 0,1: NVIDIA TITAN X (Pascal)
CUDA_HOME: /usr/local/cuda-10.2
NVCC: Cuda compilation tools, release 10.2, V10.2.8
GCC: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
PyTorch: 1.7.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.6.0 (Git Hash 5ef631a030a6f73131c77892041042805a06064f)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.2
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75
  - CuDNN 7.6.5
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON,

TorchVision: 0.8.1+cu92
OpenCV: 4.5.4
MMCV: 1.5.0
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 10.2
MMAction2: 0.24.0+26a105b

**Comment by Dai-Wenxun** [comment]
Could you please show me your `../../base/models/i3d_r50.py`?

**Comment by xiaozhucj** [comment]
This is the ../../base/models/i3d_r50.py.
# model settings
model = dict(
    type='Recognizer3D',
    backbone=dict(
        type='ResNet3d',
        pretrained2d=True,
        pretrained='torchvision://resnet50',
        depth=50,
        conv1_kernel=(5, 7, 7),
        conv1_stride_t=2,
        pool1_stride_t=2,
        conv_cfg=dict(type='Conv3d'),
        norm_eval=False,
        inflate=((1, 1, 1), (1, 0, 1, 0), (1, 0, 1, 0, 1, 0), (0, 1, 0)),
        zero_init_residual=False),
    cls_head=dict(
        type='I3DHead',
        num_classes=11,
        in_channels=2048,
        spatial_type='avg',
        dropout_ratio=0.5,
        init_std=0.01),
    # model training and testing settings
    train_cfg=None,
    test_cfg=dict(average_clips='prob'))

# This setting refers to https://github.com/open-mmlab/mmaction/blob/master/mmaction/models/tenons/backbones/resnet_i3d.py#L329-L332  # noqa: E501

**Comment by Dai-Wenxun** [comment]
Could you please debug at this [line](https://github.com/open-mmlab/mmaction2/blob/eb955825653dbbbdc8b7bb99a2d1096666e1ba27/mmaction/core/evaluation/accuracy.py#L150) to see what does the `match_array` look like?

**Comment by xiaozhucj** [comment]
match_ Array is  the format of ndarray. Then, when calculating the top-5acc, the value is as follows:
![image](https://user-images.githubusercontent.com/90119603/207803819-d59dedc1-b5bf-4f36-9a5b-85d7c5b83066.png)

The calculation code is shown in the following figure:
![Uploading image.png…]()
So the top-5acc tested here is greater than 1.

**Comment by xiaozhucj** [comment]
At that time, I still didn't understand how to display indicators during training, such as top kaccuracy

**Comment by Dai-Wenxun** [comment]
Here are my runtime variables in the frame of `top_k_accuracy`:
![image](https://user-images.githubusercontent.com/58767402/208037565-a2af483e-b8ab-453a-94e9-d34411b553f0.png)
The `match_array` should be a bool array with the shape of `(num_examples, )` rather than the shape of yours.

**Comment by Dai-Wenxun** [comment]
The training top-k accuracy is calculated at this [line[(https://github.com/open-mmlab/mmaction2/blob/master/mmaction/models/heads/base.py#L98). I think it is the problem of your `labels`, bro.

**Comment by xiaozhucj** [comment]
> 训练的 top-k 准确度是在此 [行 [(https://github.com/open-mmlab/mmaction2/blob/master/mmaction/models/heads/base.py#L98)] 处计算的。我认为这是你的问题`labels`，兄弟。

但我使用slowfast训练的时候，是可以输出top-k accuracy的。
![image](https://user-images.githubusercontent.com/90119603/208043536-b31f1ef6-81e0-4d6a-8e4a-12ea4b52394c.png)
但还是非常感谢您的帮助。
我去仔细的debug一下。

**Comment by hsp2454** [comment]
Hey, buddy, have you solved it yet

**Comment by xiaozhucj** [comment]
I haven't solved this problem. I've been busy with other things recently.

**Comment by muneebsaif** [comment]
I think they are not dividing the final result by the 5 that's why the accuracy is greater than 1 because the top 5_accuracy is 3.6300 and when we divide it by 5 then the result is 0.726.

**Comment by saadalamgir18** [comment]
thanks @muneebsaif for the answer. The issue has been resolved

**Comment by MudassirIsra** [comment]
@muneebsaif appreciatable favour 💯
